### DISCUSSION TITLE
Add whisper event

### DISCUSSION THREAD

**Discussion by FireNameFN** [PaperMC/Paper#10547] [Ideas]
### Is your feature request related to a problem?

I need to format whisper messages.

### Describe the solution you'd like.

Add `AsyncChatWhisperEvent`.

Add `AsyncChatWhisperEvent.renderer(ChatRenderer)`.
or
Add `AsyncChatWhisperEvent.senderRenderer(WhisperRenderer)`.
Add `AsyncChatWhisperEvent.receiverRenderer(WhisperRenderer)`.

### Describe alternatives you've considered.

Before I did it with PlayerCommandPreprocessEvent.
Now (after 1.19.1) if you cancel this event (`msg` command), player will be kicked on next chat message.
Kick reason: Received chat packet with missing or invalid signature.

### Other

![image](https://user-images.githubusercontent.com/37689533/182306781-bc3d9c23-95a5-460f-b617-583c143f5513.png)

![image](https://user-images.githubusercontent.com/37689533/182306947-0294821d-e0ff-4b19-b129-8b385314aa4c.png)

**Comment by MiniDigger** [comment]
fwiw, the kick on cancelling preprocess should be fixed

**Comment by FireNameFN** [comment]
> fwiw, the kick on cancelling preprocess should be fixed

Yes, it was fixed. But whisper event would be very useful.

**Comment by Machine-Maker** [comment]
There already is a `AsyncChatCommandDecorateEvent`. There is no context on what command is being run due to lack of api, but that is coming in the future.
